Frameworks/Foundation/NSKVOSupport.mm

```diff
@@ -647,15 +650,13 @@ - (void)didChangeValueForKey:(NSString*)key {
 */
 - (void)willChange:(NSKeyValueChange)changeKind valuesAtIndexes:(NSIndexSet*)indexes forKey:(NSString*)key {
     if ([self observationInfo]) {
-        NSMutableDictionary* change = [NSMutableDictionary
-            dictionaryWithObjectsAndKeys:@(changeKind), NSKeyValueChangeKindKey, indexes, NSKeyValueChangeIndexesKey, nil];
-
-        _dispatchWillChange(self, key, [ changeKind, indexes, key, notifyingObject = self, change ](_NSKVOKeyObserver * keyObserver) {
+        _dispatchWillChange(self, key, [ changeKind, indexes, key, notifyingObject = self ](_NSKVOKeyObserver * keyObserver) {
+            NSMutableDictionary* change = [NSMutableDictionary
+                dictionaryWithObjectsAndKeys:@(changeKind), NSKeyValueChangeKindKey, indexes, NSKeyValueChangeIndexesKey, nil];
             NSKeyValueChange localChangeKind = changeKind;
```

Comment:
since we're not mutating this, do we need to store it locally instead of in the capture?